### PR TITLE
handle missing xml during form processing

### PR DESCRIPTION
**Pull request by snopoke**
https://sentry.io/organizations/dimagi/issues/1121594548/

https://dimagi-dev.atlassian.net/browse/IIO-372

Redo of https://github.com/dimagi/commcare-hq/pull/24991

### CODE REVIEW

File: corehq/form_processor/models.py [hunk 1]
```diff
@@ -553,10 +553,16 @@ def get_data(self, path):
 
     @memoized
     def get_xml(self):
-        return self.get_attachment('form.xml')
+        try:
+            return self.get_attachment('form.xml')
+        except NotFound:
+            raise MissingFormXml(self.form_id)
```
Comment (millerdev) [inline corehq/form_processor/models.py]: 💯 Very happy to see this, thank you!

**Review by millerdev — APPROVED** [review]
Nice! I like this solution.

**Review by sravfeyn — APPROVED** [review]

**Comment by sravfeyn** [comment]
Restarted the two failed builds

**Comment by millerdev** [comment]
Based on its name, looks like at least one test failure may be real. Interestingly it only failed on Python 2. I'll be happy to review that test if it's an intermittent failure.

`ERROR: corehq.apps.couch_sql_migration.tests.test_migration:MigrationTestCase.test_form_with_missing_xml`

**Comment by snopoke** [comment]
@millerdev should be fixed by bcb4878e967871e5e37327ef61e5e6ecdf40f107

**Review by millerdev — APPROVED** [review]

**Pull request merged**